doc/content/physics/reactive_infiltration.md

```diff
@@ -0,0 +1,371 @@
+
+> Mathematically, it is possible that \f$ \varphi_L + \varphi_P + \varphi_S \ge 1 \f$. Physically, this implies "overflow", aka the prescribed \f$\alpha_L\f$ is larger than the available voids. Care must be taken at the macroscopic model to avoid or resolve this issue. 
+
+### Governing Equations
+
+The initial-value problem (IVP) corresponding to the constitutive model is
+\f[
+    \mathbf{r} = \begin{Bmatrix}
+        r_{\varphi_P} \\
+        r_{\varphi_S}
+    \end{Bmatrix} = \mathbf{0},
+\f]
+where \f$r_{\varphi_i}\f$ is the residual corresponding to the volume fraction of the liquid, solid and product. The residuals are defined below. Note an implicit backward-Euler time intergration scheme is used.
+
+- Production rate of the product, dictates by the diffusion of the liquid through the product to react with the solid at the product-solid interface
+\f[
+    r_{\varphi_P} = \dfrac{\alpha_{P, n+1} - \alpha_{P, n}}{t_{n+1}-t_n} - \dfrac{2}{\Omega_L}\dfrac{D}{l_c^2}\dfrac{\bar{r}_o+\bar{r}_i}{\bar{r}_o-\bar{r}_i} R_L R_S
+\f]
+Here \f$R_L(\varphi_L), R_S(\varphi_S) \in [0, 1]\f$ are the reactivity of liquid and solid, represented by a step function of the void fraction, \f$\varphi_i\f$. 
+
+- Reaction rate of the solid, with a (solid, product) reaction coefficient, \f$(k_S, k_P)\f$ respectively
+\f[
+    r_{\varphi_S} = \dfrac{\alpha_{S, n+1} - \alpha_{S, n}}{t_{n+1}-t_n} + \dfrac{k_P}{k_S} \dfrac{\alpha_{P, n+1} - \alpha_{P, n}}{t_{n+1}-t_n}
+\f]
+
+### Implementation Details
+
+The above governing equations are fairly general for a wide range of reactive infiltraion systems within this framework as long as the chemical reactions kinetics with the appropriate reaction coefficient (\f$k_i\f$), and the correct corresponding molar volume, \f$\Omega_i\f$, of the involving materials.
+
+<span style="color:red">Example</span> for determining \f$k_i\f$ and \f$\Omega_i\f$:
+- Considered a reactive infiltraion process of liquid (chemical formula \f$L_x\f$) and solid (\f$S_y\f$), with the molar mass \f$M_L\f$ and \f$M_S\f$.
+- The chemical reactions create a product with chemical formula \f$L_m S_n\f$, with a reaction coefficient \f$k_S, k_P\f$.
+\f[
+    L_x + k_S S_y \rightarrow k_P L_mS_n 
+\f]
+For chemical balance, the reaction coefficient is:
+\f{align*}
+    &\ k_S = \dfrac{xn}{ym} \quad k_P = \dfrac{x}{m} 
+\f} 
+- The material density of the liquid, solid, product is \f$\rho_L, \rho_S, \rho_P\f$.
+    
+    Then, the molar volume is:
+\f{align*}
+    &\ \Omega_L = \dfrac{x M_L}{\rho_L} \quad \Omega_S = \dfrac{y M_S}{\rho_S} \quad \Omega_P = \dfrac{mn M_LM_S}{\rho_P}
+\f}
+
+
+The chemical equation also implies that
+\f[
+    \dot{\alpha}_S = \dfrac{k_P}{k_S} \dot{\alpha}_P 
+\f]
+> Since this consitutive model considered \f$\alpha_L\f$ as the forcing function for the IVP problem. \f$\dot{\alpha}_L \ne k_P \dot{\alpha}_P\f$.
```

Comment:
```suggestion
Since this constitutive model considers \f$\alpha_L\f$ as the forcing function for the IVP problem, \f$\dot{\alpha}_L \ne k_P \dot{\alpha}_P\f$.
```
You need to remind me of the logic behind this...